samples/contrib/aws-samples/mnist-kmeans-sagemaker/mnist-classification-pipeline.py

```diff
@@ -78,187 +89,104 @@ def training_input(input_name, s3_uri):
-        output_config=process_output_config
+        container_entrypoint=[
+            "python",
+            "/opt/ml/processing/code/kmeans_preprocessing.py",
+        ],
+        input_config=[
+            processing_input(
+                "mnist_tar",
+                "s3://sagemaker-sample-data-us-east-1/algorithms/kmeans/mnist/mnist.pkl.gz",
+                "/opt/ml/processing/input",
+            ),
+            processing_input(
+                "source_code",
+                f"s3://{my_bucket_name}/mnist_kmeans_example/processing_code/kmeans_preprocessing.py",
+                "/opt/ml/processing/code",
+            ),
+        ],
+        output_config=[
+            processing_output(
+                "train_data",
+                f"s3://{my_bucket_name}/mnist_kmeans_example/",
+                "/opt/ml/processing/output_train/",
+            ),
+            processing_output(
+                "test_data",
+                f"s3://{my_bucket_name}/mnist_kmeans_example/",
+                "/opt/ml/processing/output_test/",
+            ),
+            processing_output(
+                "valid_data",
+                f"s3://{my_bucket_name}/mnist_kmeans_example/input/",
+                "/opt/ml/processing/output_valid/",
+            ),
+        ],
     )
 
     hpo = sagemaker_hpo_op(
         region=region,
-        endpoint_url=endpoint_url,
         image=train_image,
         training_input_mode=train_input_mode,
-        strategy=hpo_strategy,
-        metric_name=hpo_metric_name,
-        metric_type=hpo_metric_type,
-        early_stopping_type=hpo_early_stopping_type,
-        static_parameters=hpo_static_parameters,
-        integer_parameters=hpo_integer_parameters,
-        continuous_parameters=hpo_continuous_parameters,
-        categorical_parameters=hpo_categorical_parameters,
-        channels=hpo_channels,
+        metric_name="test:msd",
+        metric_type="Minimize",
+        static_parameters={"k": "10", "feature_dim": "784"},
+        integer_parameters=[
+            {"Name": "mini_batch_size", "MinValue": "500", "MaxValue": "600"},
+            {"Name": "extra_center_factor", "MinValue": "10", "MaxValue": "20"},
+        ],
+        categorical_parameters=[
+            {"Name": "init_method", "Values": ["random", "kmeans++"]}
+        ],
+        channels=hpoChannels,
         output_location=train_output_location,
-        output_encryption_key=output_encryption_key,
         instance_type=instance_type,
-        instance_count=instance_count,
-        volume_size=volume_size,
-        max_num_jobs=hpo_max_num_jobs,
-        max_parallel_jobs=hpo_max_parallel_jobs,
-        max_run_time=max_run_time,
-        network_isolation=network_isolation,
-        traffic_encryption=traffic_encryption,
-        spot_instance=hpo_spot_instance,
-        max_wait_time=hpo_max_wait_time,
-        checkpoint_config=hpo_checkpoint_config,
+        max_num_jobs=3,
+        max_parallel_jobs=2,
```

Comment:
I think we should have a default max here.

Reply:
This isnt a generic field and can have unintended consequences for users who dont set it. Have a default will also make it different from AWS SDK. I will leave this as-is

Reply:
Okay